### PR TITLE
Moving the walk-through into a new file

### PR DESCRIPTION
**Pull request by arschles**
Fixes https://github.com/kubernetes-incubator/service-catalog/issues/554

TODOs to add, courtesy of @rifung:

- [ ] Describe how to set up a `kubeconfig` (or new context/cluster in existing config) that talks to the service-catalog API server
- [ ] Indicate that the namespace needs to be created before starting
- [ ] Indicate that there are READMEs for the helm charts
- [ ] Document how to switch between the core API server and the service-catalog API server

### CODE REVIEW

File: docs/DEVGUIDE.md [hunk 1]
```diff
@@ -209,252 +209,13 @@ export KUBECONFIG=/home/yippee/code/service-catalog/.kubeconfig
...
-
-    $ kubectl delete brokers ups-broker
-
-And we should see that all the `ServiceClass` resources that came from that
-broker were cleaned up:
-
-    $ kubectl get serviceclasses
-    No resources found
+If you choose third party resources storage, the helm chart will not launch an etcd
+server, and instruct the API server to store all resources in the Kubernetes cluster
```
Comment (MHBauer) [inline docs/DEVGUIDE.md]: and instruct -> but will instead instruct

or some form that reads less awkwardly.

File: docs/WALKTHROUGH.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+The service catalog is conveniently packaged as a [helm](http://helm.sh/) 
+chart for installation.
+
+The chart is located in the [charts/catalog](../charts/catalog) directory in this repository, and 
+supports a wide variety of customizations which are laid out in the README.md in that
+directory. To install the service-catalog with sensible defaults, execute this command:
+
+```console
+helm install charts/catalog --name catalog --namespace catalog
```
Comment (rifung) [inline docs/WALKTHROUGH.md]: I talked to @krancour about this and he thought we should mention that you need to do

```
helm install charts/catalog --name catalog --namespace catalog --set "apiserver.service.type=NodePort"
```
if you are running in minikube.

(Currently the default is actually NodePort but from our discussion it seems that we want to change that to LoadBalancer which matches the readme)

File: docs/WALKTHROUGH.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+#### Note: accessing the service catalog
+
+Unfortunately, `kubectl` doesn't know how to speak to both the service catalog
+API server and the main Kubernetes API server without switching contexts or
+`kubeconfig` files.  For now, the best way to access the service catalog API
+server is via a dedicated `kubeconfig` file.  You can manage the kubeconfig in
+use within a directory using the `direnv` tool.
+
+----
```
Comment (rifung) [inline docs/WALKTHROUGH.md]: As discussed, we should probably add something here about how to set up the service catalog api server. I will admit I don't really know what I'm doing or if there's a better way to do all this so please feel free to ignore or change the following suggestion:

> If you want to use multiple `kubeconfig` files, you need to specify set the `KUBECONFIG` environment variable. For example, you can run
> 
> ```
> KUBECONFIG=<path_to_config> kubectl <command>
> ```
> 
> To make it less tedious, you can set up an alias, so for example, if you wanted to set your config file for the service catalog API server in your current directory you would run
> 
> ```
> alias catalogctl="KUBECONFIG=$(pwd)/kubeconfig kubectl"
> ```
> 
> To set up the service catalog api server in kubeconfig, run
> 
> ```
> catalogctl config set-credentials service-catalog-creds --username=admin --password=admin
> catalogctl config set-cluster service-catalog-cluster --server=http://${D_HOST}:${PORT}
> catalogctl config set-context service-catalog-ctx --cluster=service-catalog-cluster --user=service-catalog-creds
> catalogctl config use-context service-catalog-ctx
> ```
> 
> If you aren't using the alias then replace `catalogctl` with `KUBECONFIG=<path_to_config> kubectl`
> 
> Now you can use the `catalogctl` command whenever you want to talk to the service catalog API server.

Of course.. the issue with this is that you will have to change the rest of the walkthrough to use catalogctl instead..

Comment (pmorie) [inline docs/WALKTHROUGH.md]: Strongly do not want catalogctl.  Use direnv, that's what it's for.

Comment (rifung) [inline docs/WALKTHROUGH.md]: =[

I admit I hadn't heard about direnv until I read the walkthrough but I don't understand how you would add it to your workflow. Doesn't it force you to change directories whenever you want to change contexts? The majority of the walkthrough has the user inside the service-catalog directory so I must be missing something..

In any case perhaps it'd be useful to elaborate more on direnv for those of us who haven't used it before and we still need something on how to add the context in the first place.

**Comment by arschles** [comment]
closing in favor of #607

**Pull request closed without merging**